### PR TITLE
Fix `tls.createSecurePair` deprecation warning

### PR DESCRIPTION
**Pull request by arthurschreiber**
This is based on the great work done by @joux3 over in #689. 🙇 

I forked `duplexpair` into a `native-duplexpair`, a version that uses the `Duplex` stream exposed by Node.js' built-in `stream` module, instead of the `readable-stream` module. This solves an issue where the `duplexpair` usage on Node.JS versions before the 8.x line would cause a crash.

### CODE REVIEW

**Comment by arthurschreiber** [comment]
A few additional notes - I'm not happy about forking `duplexpair`, but for as long as we support Node.js 6.x, this is what I believe to be the best solution.

Once we drop support for Node.js 6.x, we can switch over to the original `duplexpair` module. 👍

**Comment by LiamDotPro** [comment]
Four years, Is it finally time for the depreciation warning to be depreciated, hold me boys.. It's happening 💯

**Comment by ovikholt** [comment]
How is this coming along?

**Comment by LiamDotPro** [comment]
@arthurschreiber Are there any updates to share on this? When can we expect this to be merged?

File: src/message-io.js [hunk 1]
```diff
@@ -83,47 +86,33 @@ module.exports = class MessageIO extends EventEmitter {
...
+    const securePair = this.securePair = {
+      cleartext: tls.connect({
+        socket: duplexpair.socket1,
+        secureContext: credentials,
+        rejectUnauthorized: !trustServerCertificate
+      }),
+      encrypted: duplexpair.socket2
+    };
 
-        // Verify that server's identity matches it's certificate's names
```
Comment (arthurschreiber) [inline src/message-io.js]: By switching to `tls.connect`, we no longer need to perform server verification ourselves, and can instead make use of the `rejectUnauthorized` option. 🎉

File: src/message-io.js [hunk 2]
```diff
@@ -83,47 +86,33 @@ module.exports = class MessageIO extends EventEmitter {
...
-    // Forward the error to the socket so the connection gets properly cleaned up.
-    this.securePair.cleartext.on('error', (err) => {
-      this.socket.destroy(err);
-    });
-
-    // On Node >= 0.12, the encrypted stream automatically starts spewing out
-    // data once we attach a `data` listener. But on Node <= 0.10.x, this is not
-    // the case. We need to kick the cleartext stream once to get the
-    // encrypted end of the secure pair to emit the TLS handshake data.
-    this.securePair.cleartext.write('');
```
Comment (arthurschreiber) [inline src/message-io.js]: By switching to `tls.connect`, we no longer need to "kick" the secure socket to start performing the handshake - it's all handled automatically. 👍 (Also, the comment here was severely out-of-date. 🤷‍♂️)

**Comment by LiamDotPro** [comment]
@arthurschreiber Legendary.

**Pull request merged**